test/language/module-code/early-dup-top-function.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2021 Chengzhong Wu. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+es6id: 10.2.1
+description: >
+    It is a Syntax Error if the LexicallyDeclaredNames of ModuleItemList
+    contains any duplicate entries.
+    At the top level of a Module, function declarations are treated like
+    lexical declarations rather than like var declarations.
```

Comment:
This isn't true. The grammar that makes this test correct are: 

_ModuleItemList_ :
&nbsp;&nbsp;&nbsp;&nbsp;_ModuleItem_
&nbsp;&nbsp;&nbsp;&nbsp;_ModuleItemList ModuleItem_

_ModuleItem_ :
&nbsp;&nbsp;&nbsp;&nbsp;_StatementListItem_


_StatementListItem_ :
&nbsp;&nbsp;&nbsp;&nbsp;_Declaration_

_Declaration_ :
&nbsp;&nbsp;&nbsp;&nbsp;_HoistableDeclaration_


_HoistableDeclaration_ :
&nbsp;&nbsp;&nbsp;&nbsp;_FunctionDeclaration_
&nbsp;&nbsp;&nbsp;&nbsp;_AsyncFunctionDeclaration_


However, _HoistableDeclaration_ also includes _GeneratorDeclaration_ and _AsyncGeneratorDeclaration_, so this test is incomplete because no code matching those productions is being tested.

Reply:
I was referring to the Note 3 at https://tc39.es/ecma262/#sec-static-semantics-lexicallydeclarednames. That is the only place (there are several notes similar to this one in the section) I can find that distinguishes `FunctionDeclaration` and others to be lexically scoped declarations. Is there any expected structural format of notes reference?

The grammar `Declaration` and `HoistableDeclaration` does not claim themselves as lexically scoped declarations. They are var scoped declarations in scripts. Please correct me if I'm not getting correct.